app/code/Magento/NewsletterApi/Exception/AlreadySubscribedException.php

```diff
@@ -0,0 +1,14 @@
+<?php
+/**
```

Comment:
> On the other hand, in the service contract it might be better to be a little more generic while in the implementation itself the integrator can then introduce more specialized exceptions, that are extended from the more generic exception

`AlreadySubscribedException` _is_ the most generic exception you can throw from the Newsletter service contract. 

Furthermore, we, as clients of the API, cannot use the _implementation's_ exception, like you said, as we would break the contract. We would always have to catch (and use) some `LocalizedException`, thus catching more exception than we should.